src/textual/widget.py

```diff
@@ -1581,24 +1612,58 @@ def scroll_end(
             easing: An easing method for the scrolling animation. Defaults to "None",
                 which will result in Textual choosing the configured default scrolling easing function.
             force: Force scrolling even when prohibited by overflow styling. Defaults to `False`.
-
-        Returns:
-            True if any scrolling was done.
-
         """
         if speed is None and duration is None:
             duration = 1.0
-        return self.scroll_to(
-            0,
-            self.max_scroll_y,
+
+        # In most cases we'd call self.scroll_to and let it handle the call
+        # to do things after a refresh, but here we need the refresh to
+        # happen first so that we can get the new self.max_scroll_y (that
+        # is, we need the layout to work out and then figure out how big
+        # things are). Because of this we'll create a closure over the call
+        # here and make our own call to call_after_refresh.
+        def _lazily_scroll_end():
```

Comment:
Typing and docstring please.

Reply:
I wouldn't have thought we'd docstring hidden closures, but sure thing.

Reply:
Done.